src/cryptoadvance/specter/services/extension_gen.py

```diff
@@ -46,13 +48,20 @@ def __init__(
         self.base_path = base_path
         self.org = ext_org
         self.id = ext_id
+        self.encrypted_userdata = encrypted_userdata
+        self.custom_version = custom_version
         self.isolated_client = isolated_client
         self.devicename = devicename
         self.author = author
         self.author_email = email
         vc = VersionChecker()
         version = vc._get_current_version()
-        if version == "custom":
+        if self.custom_version != "":
```

Comment:
Please have a look how `branch` and `version` is used. the feature i mostly spoke about was using specific branches (tags) as the basis for the templating [here](https://github.com/cryptoadvance/specter-desktop/blob/master/src/cryptoadvance/specter/services/extension_gen.py#L66).
Yes, the `self.version` is used for specific templates in order to render out the version this extension is dependent on. But the branch variable specifies which templates are used: [branches (or tags, doesn't matter) in the specterext-dummy-repo](https://github.com/cryptoadvance/specterext-dummy/tags). You can rename "branch" to "tag" which would make more sense.